_style/control-structures.md

```diff
@@ -41,63 +60,84 @@ Remember the following guidelines:
 
 <!-- necessary to separate the following example from the above bullet list -->
 
-    val news = if (foo)
-      goodNews()
-    else
-      badNews()
-
-    if (foo) {
-      println("foo was true")
-    }
-
-    news match {
-      case "good" => println("Good news!")
-      case "bad" => println("Bad news!")
-    }
+{% tabs control_structures_2 class=tabs-scala-version%}
+{% tab 'Scala 2' for=control_structures_2 %}
+```scala
+val news = if (foo)
+  goodNews()
+else
+  badNews()
+
+if (foo) {
+  println("foo was true")
+}
+
+news match {
+  case "good" => println("Good news!")
+  case "bad" => println("Bad news!")
+}
+```
+{% endtab %}
+{% tab 'Scala 3' for=control_structures_2 %}
+```scala
+val news = if foo then
+  goodNews()
+else
+  badNews()
+
+if foo then
+  println("foo was true")
+
+news match 
+  case "good" => println("Good news!")
+  case "bad" => println("Bad news!")
+```
+{% endtab %}
+{% endtabs %}
 
 ## Comprehensions
 
 Scala has the ability to represent `for`-comprehensions with more than
 one generator (usually, more than one `<-` symbol). In such cases, there
 are two alternative syntaxes which may be used:
 
-    // wrong!
-    for (x <- board.rows; y <- board.files)
-      yield (x, y)
-
-    // right!
-    for {
-      x <- board.rows
-      y <- board.files
-    } yield (x, y)
+{% tabs control_structures_3 class=tabs-scala-version%}
+{% tab 'Scala 2' for=control_structures_3 %}
+```scala
+// wrong!
+for (x <- board.rows; y <- board.files)
+  yield (x, y)
+
+// right!
+for {
+  x <- board.rows
+  y <- board.files
+} yield (x, y)
+```
+{% endtab %}
+{% tab 'Scala 3' for=control_structures_3 %}
+```scala
+// wrong!
+for x <- board.rows; y <- board.files
+  yield (x, y)
+
+// right!
+for
+  x <- board.rows
+  y <- board.files
+yield (x, y)
+```
+{% endtab %}
+{% endtabs %}
 
 While the latter style is more verbose, it is generally considered
 easier to read and more "scalable" (meaning that it does not become
 obfuscated as the complexity of the comprehension increases). You should
 prefer this form for all `for`-comprehensions of more than one
 generator. Comprehensions with only a single generator (e.g.
-`for (i <- 0 to 10) yield i`) should use the first form (parentheses
+`for i <- 0 to 10 yield i`) should use the first form (parentheses
 rather than curly braces).
 
-The exceptions to this rule are `for`-comprehensions which lack a
-`yield` clause. In such cases, the construct is actually a loop rather
-than a functional comprehension and it is usually more readable to
-string the generators together between parentheses rather than using the
-syntactically-confusing `} {` construct:
```

Comment:
I suggest to drop this since the arguments for using one-generator-per-line syntax hold no matter whether the next keyword is `do` or `yield`